### PR TITLE
feat: 10 new features

### PR DESCRIPTION
**Pull request by chafnan**
- added target device for page selection
- cut, paste, delete menu for buttons
- added font size change for button text
- added font color choice
- added font types
- added keyboard shortcuts for cut, copy, paste
- text align - left, center, right
- new line support with `\n` character in text
- added feedback to button press
- added option to enable/disable feedback and custom image

### CODE REVIEW

**Comment by masc2279** [comment]
Works great on multi stream decks ... Thanks !

**Comment by masc2279** [comment]
Added Delete button, Cut Button, Paste Button, and multi Paste button feature to the single and multi device code

**Comment by dodgyrabbit** [comment]
@chafnan & @masc2279  Thanks for sharing your contributions!
Things have been a bit busy, but I'd like to see how we can incorporate this work.

This MR has become a bit unwieldy.  It would be more practical if individual features are separate PRs. This makes it easier to review and allows to merge items independently while others may need a bit of rework.

I noticed that you imported tkinter. This would create a dependency on a whole other UI framework. Are you able to remove this dependency and stick to PySide2 only? Many users may not have, or want, this extra dependency.  The window styling also appears to be quite different between the two systems, which looks odd.

The other thought is that the main UI is getting a bit cluttered. It may make sense to instead use the PySide2 font dialog.
https://doc.qt.io/archives/qtforpython-5.14/PySide2/QtWidgets/QFontDialog.html I.e. have a button that brings up font settings so that it does not take up so much of the configuration UI.

FYI: I have another branch that may compliment your work quite well. It uses the _actual button image (including text)_ in the button UI. This way you can see the color, font etc on screen that matches what is on the StreamDeck.

**Comment by chafnan** [comment]
It's an interesting conundrum to get in to develop a lot of features and not have any merged. It is the only reason why we just put them all in one branch.

That being said, we are ready to help out to get this merged and share all these great features with the user base. My suggestion would be that we remove the tkinter dependency, you merged the branch, and then we will work on refactoring the features to work with your button image class.

Let me know what you think.

**Comment by dodgyrabbit** [comment]
Yes, I think it would be very helpful to refactor tkinter dependency out.  Generally speaking, smaller PRs are always preferred over large ones.  However, I appreciate that your _intention_ was to make this easy! One way or another, we'll get this work merged in don't worry about that.

When your refactor is ready, I'll merge this into a separate branch. From there I can cherry pick into separate feature branches and merge (or some other strategy, I'll have to see).

**Separate topic:**
Have you looked at using the fonts installed on the system instead of redistributing font files?
The preference would be that we use PySide QFontDialog which would give users a wider array of fonts and also avoid redistributing font files. Did a bit of reading today - main issue I see is that there may be a bit tricky to move the QFont or QFontInfo (PySide object) to PIL ImageFont.truetype. https://stackoverflow.com/a/64729615/192815

File: streamdeck_ui/gui.py [hunk 1]
```diff
@@ -3,6 +3,8 @@
 import shlex
 import sys
 import time
+import tkinter as tk
```
Comment (OWKenobi) [inline streamdeck_ui/gui.py:6]: tkinter is not installed on ubuntu and breaks the code :-/

**Comment by dodgyrabbit** [comment]
Doing some cleanup. Main branch have substantially diverged, no response in a long time.  Nothing to do but close.

**Pull request closed without merging**